src/components/Select2.js

```diff
@@ -69,6 +69,14 @@ export default class Select2 extends Component {
       this.initSelect2(false);
     }
 
+    const { options } = this.props;
+    if (!shallowEqualFuzzy(prevProps.options, options)) {
+      if (typeof options.dropdownParent === 'string') {
+        options.dropdownParent = $(options.dropdownParent);
```

Comment:
Maybe make a separate method `prepareOptions`?
To avoid duplicate code:
```js
if (typeof options.dropdownParent === 'string') {
  options.dropdownParent = $(options.dropdownParent);
```